libs/langgraph/src/managed/shared_value.ts

```diff
@@ -0,0 +1,123 @@
+import { RunnableConfig } from "@langchain/core/runnables";
+import { BaseStore, type Values } from "../store/base.js";
+import {
+  ChannelKeyPlaceholder,
+  ConfiguredManagedValue,
+  ManagedValue,
+  ManagedValueParams,
+  WritableManagedValue,
+} from "./base.js";
+import { CONFIG_KEY_STORE } from "../constants.js";
+import { InvalidUpdateError } from "../errors.js";
+
+type Value = Record<string, Values>;
+type Update = Record<string, Values | null>;
+
+export interface SharedValueParams extends ManagedValueParams {
+  scope: string;
+  key: string;
+}
+
+export class SharedValue extends WritableManagedValue<Value, Update> {
+  scope: string;
+
+  store: BaseStore | null;
+
+  ns: string | null;
+
+  value: Value = {};
+
+  constructor(config: RunnableConfig, params: SharedValueParams) {
+    super(config, params);
+    this.scope = params.scope;
+    this.store = config.configurable?.[CONFIG_KEY_STORE] || null;
+
+    if (!this.store) {
+      this.ns = null;
+    } else if (config.configurable?.[this.scope]) {
+      const scopeValue = config.configurable[this.scope];
+      let scopedValueString = "";
+      try {
+        scopedValueString =
+          typeof scopeValue === "string"
+            ? scopeValue
+            : JSON.stringify(scopeValue);
+      } catch (_) {
```

Comment:
When would this fail?

Reply:
Good catch, I don't think it will. removed